### DISCUSSION TITLE
API/Schema first support?

### DISCUSSION THREAD

**Discussion by tigerinus** [smallrye/smallrye-graphql#1462] [Q&A] [answered]
Hello,

Our team embraces API/Schema first development approach. We come up with GraphQL schemas first, and use codegen tools for the following development. Our early feasibility study of `smallrye-graphql` concluded with very positive feedback in terms of transitioning to Java native world, but currently blocked due to that it's *code first* approach. 

Does anyone have any recommendation / thoughts regarding API/Schema first approach with `smallrye-graphql`? Are we at dead end or is there anything we can do?

Thanks!

**Comment by phillip-kruger** [comment, marked as answer]
It's not smallrye-graphql that gives to a transitioning to Java Native, but rather Quarkus, that use smallrye-graphql. You can still use a code first approach in Quarkus. Create your schema and generate the relevant code. Include this in a Quarkus App, and then include the vert-graphql extension, that will pull in graphql-java. SmallRye GraphQL is a code first library. It generates the Schema and gives you an easy way to create endpoints. There is not much value to add when you provide the schema and generate the code.
